services/pull/check.go

```diff
@@ -98,15 +98,19 @@ func CheckPullMergable(stdCtx context.Context, doer *user_model.User, perm *acce
 		}
 
 		if err := CheckPullBranchProtections(ctx, pr, false); err != nil {
-			if models.IsErrDisallowedToMerge(err) {
-				if force {
-					if isRepoAdmin, err2 := access_model.IsUserRepoAdmin(ctx, pr.BaseRepo, doer); err2 != nil {
-						return err2
-					} else if !isRepoAdmin {
-						return err
-					}
-				}
-			} else {
+			if !models.IsErrDisallowedToMerge(err) {
+				log.Error("Error whilst checking pull branch protection for %-v: %v", pr, err)
+				return err
+			}
+
+			if !force {
+				return err
+			}
+
+			if isRepoAdmin, err2 := access_model.IsUserRepoAdmin(ctx, pr.BaseRepo, doer); err2 != nil {
+				log.Error("Unable to check if %-v is a repo admin in %-v: %v", doer, pr.BaseRepo, err2)
+				return err2
+			} else if !isRepoAdmin {
```

Comment:
What I just noticed:
Under what conditions is `models.ErrDisallowedToMerge` thrown for a repo admin?

Reply:
Plenty of times:

https://github.com/go-gitea/gitea/blob/c8093a13ff89834d587556a292c4c8cbfdb700a5/services/pull/merge.go#L758-L815

* When the CI fails.
* When there is not enough approvals.
* When there is a rejecting review.
* Missing official review requests.
* The branch needs updating.
* When a protected file is changed.